### DISCUSSION TITLE
Validator compiler

### DISCUSSION THREAD

**Discussion by onosendi** [fastify/fastify#4577] [Q&A] [answered]
What's the proper way to set `validatorCompiler` for all routes?

**Comment by metcoder95** [comment, marked as answer]
It should work by setting it at the root/parent context of your routes, and it will be applied to the child ones.

**Comment by Eomm** [comment]
It is used to replace `ajv` with any other framework you need

Here is an example with the `joi` implementation: https://github.com/Eomm/joi-compiler
